### ISSUE TITLE
sqlite outdated

### ISSUE THREAD

**Issue by MikeMoy** [tekartik/sqflite#751]
The most recent revision of sqflite 2.0.1 is using an old version of sqlite that has since been updated 34 times since the year 2018. 

The most recent revision of sqflite 2.0.1 is using sqlite 3.22.0 which was published 22 Febuary 2018.

However sqlite has since been revised many times with sqflite 3.37.0 being published 27 November 2021.

Modern features of sqlite are thus missing in sqflite 2.0.1 such as the ability to enforce strict data types when creating database tables.

**Comment by alextekartik** [comment]
`sqflite` uses the SQLite available on the platform. It does not ship/bundle any additional SQLite library. 

Unfortunately the version of SQLite depends on the OS version.

You could get a more recent version using [`sqflite_common_ffi`](https://pub.dev/packages/sqflite_common_ffi).

You could then add [`sqlite3_flutter_libs`](https://pub.dev/packages/sqlite3_flutter_libs) for ios/android or include your own
sqlite shared library for desktop or mobile (one for each platform).

https://github.com/tekartik/sqflite/blob/master/sqflite/doc/version.md

**Comment by MikeMoy** [comment]
Yes using[ sqflite_common_ffi](https://pub.dev/packages/sqflite_common_ffi) I got the most recent version of sqflite 3.37.0.

However when issuing the command to create a STRICT database table it creates a malformed table that is corrupted. 


```
import 'package:sqflite_common/sqlite_api.dart';
import 'package:sqflite_common_ffi/sqflite_ffi.dart';

  sqfliteFfiInit();

  var databaseFactory = databaseFactoryFfi;
  var db = await databaseFactory.openDatabase(inMemoryDatabasePath);
  await db.execute('''
  CREATE TABLE Product (
      id INTEGER PRIMARY KEY,
      title TEXT
  ) STRICT
  ''');
```
